packages/@ember/-internals/metal/lib/tracked/debugging.ts

```diff
@@ -0,0 +1,110 @@
+import { Tag, CONSTANT_TAG, UpdatableTag } from '@glimmer/reference';
+
+import { Tracker } from './tracker';
+
+type Option<T> = T | null;
+
+Ember.EMBER_DEBUG = {};
+
+let TAG_MAP = new Map<Tag | UpdatableTag, any>();
+
+interface TrackerHistory {
+  property: string;
+  value: any;
+  tracker: Tracker;
+}
+
+interface DebugTracking {
+  history: TrackerHistory[];
+  stacks: Set<Tracker>;
+}
+
+interface TrackerSnapshot {
+  tags: Array<Tag | UpdatableTag>;
+}
+
+interface TrackerProperty {
+  propertyName: string;
+  objectName: string;
+  objectRef: object;
+  tag: Tag | UpdatableTag;
+}
+
+Ember.EMBER_DEBUG.TRACKING = {
+  history: [],
+  stacks: new Set<Tracker>(),
+} as DebugTracking;
+
+function normalizeTrackerEntry(tag: any) {
+  // it looks like there are two types of data on a tracker's tag list.
+  // one type has the shape (which maybe isn't a tag?):
+  //   key: string
+  //   ref: object
+  //   tag: object
+  // the other is an actual Tag,
+  //   but with a private secret property _propertyName
+
+  let hostObject = TAG_MAP.get(tag);
```

Comment:
this is always undefined -- I _can_ access the object on tags who have a `ref` -- which, I think are tags that are from component invocations.